content/blog/embrace-kubernetes-part1/index.md

```diff
@@ -0,0 +1,96 @@
+
+Secondly, the actual interface required to deploy something with Kubernetes is also complex. It can be many thousands of lines of YAML, which often needs to be templated and manipulated with mechanisms that YAML was never designed to use. Take a look at [Why the **** are we templating YAML?](https://leebriggs.co.uk/blog/2019/02/07/why-are-we-templating-yaml.html) for a strong opinion on this matter.
+
+Finally, when something goes wrong, you need a subject matter expert to fix it. Because Kubernetes gives you such a high degree of abstraction, it’s very difficult to debug and most people will be unfamiliar with what’s going on under the hood. You need someone who’s really knowledgeable when there’s trouble. In fact, you probably need two people. You’re too vulnerable if that person leaves the company and you’re completely unreasonable if you can’t even let that person take a vacation.
+
+#### Can you afford the technical debt?
+
+Because you’re dedicating several people to Kubernetes, you may be increasing your technical debt in other areas. Do you still have people available to handle bugs and other support issues?
+
+#### Can you afford the extra infrastructure?
+
+As with any large migration, moving from your current system to Kubernetes needs to happen cautiously. You’re not simply going to destroy your existing infrastructure. Instead, you’ll probably want to adopt a side-by-side approach and slowly move your applications across. In other words, for quite a while, you’ll need to support two infrastructures rather than one.
+
+### Is Kubernetes More than You Need?
+
+Many companies, and startups in particular, don’t want to plan for their current user base. They want an infrastructure, such as Kubernetes, that they think will support the million users they see in their future. Does this sort of planning make sense?
+
+Perhaps it’s better to spend your time understanding what your business value is and building up that value. For infrastructure, when you’re starting out, consider off-the-shelf options such as Heroku or serverless compute.
+
+Architectures have to be shaped over time. You can't simply create your infrastructure once and think you’re finished.Whatever infrastructure you adopt, you need to be able to change and adapt. In five years time, who knows what will be available? Don’t lock yourself into an architecture prematurely.
+
+Learn More
+This blog post talked about the many issues to consider before you dive headlong into a Kubernetes project. If you’re curious about just what a small project might look like (using your favorite programming language, whatever that is), check out Pulumi’s Get Started with Kubernetes.
+
+## Learn More
```

Comment:
@jaxxstorm I find the conclusion of the article a little strange ... feels like we should have something like

"Ok, those were a lot of reasons you _shouldn't_ adopt Kubernetes. But there are plenty of reasons you _should_. Stay tuned for part 2 where we'll dive deeper into that topic. In the meantime, if you've already decided, .... [HERE IS THE CTA TO START WITH PULUMI]"

Reply:
Also seems like a great  opportunity to plug https://www.pulumi.com/docs/guides/crosswalk/kubernetes/?

Reply:
updated to:

This blog post talked about the many issues to consider before you dive headlong into a Kubernetes project. However, there are just as many reasons to choose Kubernetes for deploying your modern applications. Stay tuned for part 2 where we'll dive deeper into that topic. If you’re curious about just what a small project might look like (using your favorite programming language, whatever that is), check out Pulumi’s [Get Started with Kubernetes]({{< relref "/docs/get-started/kubernetes" >}}) and [Pulumi Crosswalk for Kubernetes]({{< relref "/docs/guides/crosswalk/kubernetes" >}}), playbooks with built in best practices that make Kubernetes accessible to everyone.